### PR TITLE
Deprecate the FairScale integration

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

See title.

I haven't added deprecation tests as they are extensively covered in the sharded tests.
All deprecations use the exact same message, so there's no need to check whether it has been shown already. (warnings do this by default)

### Does your PR introduce any breaking changes? If yes, please list them.

None

cc @borda @tchaton @awaelchli

### CODE REVIEW

File: src/pytorch_lightning/strategies/sharded.py [hunk 1]
```diff
@@ -40,6 +41,15 @@ class DDPShardedStrategy(DDPStrategy):
     strategy_name = "ddp_sharded"
     _REDUCE_BUFFER_SIZE_DEFAULT: int = 2**23  # 8M
 
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        rank_zero_deprecation(
+            "PyTorch Lightning's sharded implementation using FairScale has been deprecated in v1.9.0 and will be"
+            " removed in v2.0.0. You can try using the `Trainer(strategy='fsdp_native')` instead."
+            " The difference is that native FSDP uses PyTorch's implementation and the current strategy uses"
+            " FairScale's implementation (which was upstreamed to PyTorch). After removal, `strategy='fsdp'` will use"
+            " the native version by default."
```
Comment (awaelchli) [inline src/pytorch_lightning/strategies/sharded.py:50]: Great, this aligns well with Fabric

File: docs/source-pytorch/advanced/model_parallel.rst [hunk 2]
```diff
@@ -171,178 +168,25 @@ these benefits in multi-GPU setups are almost free and throughput scales well wi
...
-            self.block_1 = checkpoint_wrapper(nn.Sequential(nn.Linear(32, 32), nn.ReLU()))
-            self.block_2 = nn.Linear(32, 2)
-
-----
-
-.. _fully-sharded-native-training:
-
 ******************************
-PyTorch Fully Sharded Training
+Fully Sharded Training
```
Comment (awaelchli) [inline docs/source-pytorch/advanced/model_parallel.rst:184]: needs to adjust the underline for the title

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Comment by carmocca** [comment]
There's an issue with standalone tests hanging, marking as a draft...

**Comment by carmocca** [comment]
The hang is caused by
```python
with pytest.deprecated_call():
    trainer = Trainer()
trainer.fit()
```
when multiple processes are used

because the deprecation message is only shown on rank 0, the other ranks fail the `deprecate_call` and then `fit` hangs after because some processes didn't make it.

We would need a custom `rank_zero_deprecated_call` testing utility to avoid this.

**Pull request merged**